toolchain/check/node_block_stack.h

```diff
@@ -76,10 +70,27 @@ class NodeBlockStack {
   // Prints the stack for a stack dump.
   auto PrintForStackDump(llvm::raw_ostream& output) const -> void;
 
-  auto empty() const -> bool { return stack_.empty(); }
-  auto size() const -> size_t { return stack_.size(); }
+  auto empty() const -> bool { return size() == 0; }
+  auto size() const -> size_t { return size_; }
 
  private:
+  struct StackEntry {
+    StackEntry() { content.reserve(32); }
```

Comment:
Please add comments here about how this constant was chosen, at least for future understanding, even if it's arbitrary.

I'm used to 60-line functions being common, so this seems small -- though maybe with flow control it breaks down closer to this size, I'm not really digging into details.

I assume the management means any growth from a large function is permanent and deliberate, but could have some unintended consequences...

Reply:
Added a comment saying this is arbitrary and should be replaced by something based on an actual measurement.